lib/connection.js

```diff
@@ -447,16 +447,20 @@ class Connection extends EventEmitter {
     }
 
     const applicationName = message.handshake[1];
-    const applicationVersion = message.handshake[2];
+    this.requestedVersion = message.handshake[2];
     const application = this.server._getApplication(
-      applicationName, applicationVersion
+      applicationName, this.requestedVersion
     );
 
     if (!application) {
       this._handshakeError(errors.ERR_APP_NOT_FOUND);
       return;
     }
 
+    if (!this.requestedVersion) {
+      this.requestedVersion = semver.major(application.version).toString();
```

Comment:
In case when there is only one app without version and no version specified upon handshake call this still seems to call semver.major on undefined, or have I missed something?

Reply:
Fixed.